### ISSUE TITLE
return sequence=true or return sequence is false

### ISSUE THREAD

**Issue by jjsr** [cerlymarco/MEDIUM_NoteBook#38]
Sorry, sir for asking my query like this I urge you to please write a medium article on attention too this is about your StackOverflow answer of this(I have a low reputation for post  a comment there) -
https://stackoverflow.com/questions/62948332/how-to-add-attention-layer-to-a-bi-lstm/62949137#62949137

Sir using this I have implemented the custom Attention layer as well. Sir What is the difference between return_sequence =True and return_sequence=False. Is return sequence =True is able to return the weighted sum of attention vectors ??  I am trying to implement attention of [this paper](https://www.scitepress.org/Papers/2021/102842/102842.pdf). As per paper Context vector is weighted sum of softmax attention vector at given time step . Is with return sequence = True I am able to get Weighted sum of the context vector of all time step of Do I need to change it . Demo attention code is `from tensorflow.keras.layers import Layer
from tensorflow.keras import backend as K

class Attention(Layer):
    
    def __init__(self, return_sequences=True):
        self.return_sequences = return_sequences
        super(Attention,self).__init__()
        
    def build(self, input_shape):
        
        self.W=self.add_weight(name="att_weight", shape=(input_shape[-1],1),
                               initializer="normal")
        self.b=self.add_weight(name="att_bias", shape=(input_shape[1],1),
                               initializer="zeros")
        
        super(Attention,self).build(input_shape)
        
    def call(self, x):
        
        e = K.tanh(K.dot(x,self.W)+self.b)
        a = K.softmax(e, axis=1)
        output = x*a
        
        if self.return_sequences:
            return output
        
        return K.sum(output, axis=1)`
Please help me in this regard sir I am stuck over it for days . Thanks in advance

**Comment by cerlymarco** [comment]
Hi,

I think this is not the right place to ask this question. However, I didn't read the paper, but what I can say is that the attention u reported is a self-attention mechanism.

With return_sequences=True, u get the 3D input vector multiplied by the attention weights (which is again a 3D tensor).

With return_sequences=False, u get the weighted sum between the input and the attention weights (which is a 2D tensor).

In the end, if u support the project, don't forget to leave a star.
All the best

**Comment by jjsr** [comment]
Thanks a lot, sir for your kind reply. This will help me a lot sir. Giving a star is my utmost pleasure.
